bindings/nodejs/src/lib.rs

```diff
@@ -22,94 +22,230 @@ use std::str;
 use chrono::DateTime;
 use chrono::NaiveDateTime;
 use chrono::Utc;
-use opendal::services;
-use futures::prelude::*;
+use futures::{TryStreamExt};
 use napi::bindgen_prelude::*;
 
 #[napi]
-pub struct OperatorFactory {}
+pub struct Memory {}
 
 #[napi]
-impl OperatorFactory {
+impl Memory {
+    #[napi(constructor)]
+    pub fn new() -> Self {
+        Self {}
+    }
+
     #[napi]
-    pub fn memory() -> Result<Operator> {
-        let op = opendal::Operator::create(services::Memory::default())
-            .unwrap()
-            .finish();
+    pub fn build(&self) -> Operator {
+        self.make_operator()
+    }
+}
 
-        Ok(Operator::new(op))
+trait OperatorBuilder {
+    fn make_operator(&self) -> Operator;
+}
+
+impl OperatorBuilder for Memory {
+    fn make_operator(&self) -> Operator {
+        Operator(opendal::Operator::create(opendal::services::Memory::default()).unwrap().finish())
     }
 }
 
-#[allow(dead_code)]
 #[napi]
-pub struct ObjectMeta {
-    pub location: String,
-    pub last_modified: i64,
-    pub size: u32
+pub struct Operator(opendal::Operator);
+
+#[napi]
+impl Operator {
+    #[napi]
+    pub fn object(&self, path: String) -> DataObject {
+        DataObject(self.0.object(&path))
+    }
 }
 
 #[napi]
-pub struct Operator {
-    inner: opendal::Operator
+pub enum ObjectMode {
+    /// FILE means the object has data to read.
+    FILE,
+    /// DIR means the object can be listed.
+    DIR,
+    /// Unknown means we don't know what we can do on this object.
+    Unknown,
 }
 
+#[allow(dead_code)]
 #[napi]
-impl Operator {
-    pub fn new(op: opendal::Operator) -> Self {
-        Self { inner: op }
+pub struct ObjectMetadata(opendal::ObjectMetadata);
+
+#[napi]
+impl ObjectMetadata {
+    /// Mode of this object.
+    #[napi(getter)]
+    pub fn mode(&self) -> ObjectMode {
+        match self.0.mode() {
+            opendal::ObjectMode::DIR => ObjectMode::DIR,
+            opendal::ObjectMode::FILE => ObjectMode::FILE,
+            opendal::ObjectMode::Unknown => ObjectMode::Unknown,
+        }
+    }
+
+    /// Content-Disposition of this object
+    #[napi(getter)]
+    pub fn content_disposition(&self) -> Option<String> {
+        self.0.content_disposition().map(|s| s.to_string())
+    }
+
+    /// Content Length of this object
+    #[napi(getter)]
+    pub fn content_length(&self) -> Option<u32> {
+        u32::try_from(self.0.content_length()).ok()
+    }
+
+    /// Content MD5 of this object.
+    #[napi(getter)]
+    pub fn content_md5(&self) -> Option<String> {
+        self.0.content_md5().map(|s| s.to_string())
+    }
+
+    /// Content Range of this object.
+    #[napi(getter)]
+    pub fn content_range(&self) -> Option<Vec<u32>> {
+        let content_range = self.0
+            .content_range()
+            .unwrap_or_default();
+        let range = content_range.range().unwrap_or_default();
+        Some(vec![
```

Comment:
We can expose this API as a string.

NOTE: `Option<T>` has strict meaning in OpenDAL, it's incorrect to `unwrap_or_default` a field of ObjectMetadata and always return `Some(T)` to users.

Reply:
As a string? As a number array, how should we represent it using string? 
Like `1,2,3`?

Maybe also a BigInt array.

> NOTE: Option<T> has strict meaning in OpenDAL, it's incorrect to unwrap_or_default a field of ObjectMetadata and always return Some(T) to users.

SGTM.

Reply:
Let's remove this api first. Content range is a bit special that it's content related to users request.